plantuml/plantuml-server
# Update JUnit Test Classes and there dependencies

This PR looks still very huge. However, this is due to the exchange of the test framework.
This PR is one part of #189.

### File changes summary

```
pom.xml                                                             |  22 +--
src/main/java/net/sourceforge/plantuml/servlet/OldProxyServlet.java |  68 +++----
src/main/java/net/sourceforge/plantuml/servlet/ProxyServlet.java    |   1 +
src/main/webapp/index.jsp                                           |   2 +-
src/test/java/net/sourceforge/plantuml/servlet/ServerUtils.java     |  38 ----
src/test/java/net/sourceforge/plantuml/servlet/TestAsciiArt.java    |  24 +--
src/test/java/net/sourceforge/plantuml/servlet/TestCharset.java     |  43 +++--
src/test/java/net/sourceforge/plantuml/servlet/TestCheck.java       |  40 ++--
src/test/java/net/sourceforge/plantuml/servlet/TestEPS.java         |  25 +--
src/test/java/net/sourceforge/plantuml/servlet/TestForm.java        | 294 ++++++++++++++++++------------
src/test/java/net/sourceforge/plantuml/servlet/TestImage.java       |  84 ++++-----
src/test/java/net/sourceforge/plantuml/servlet/TestLanguage.java    |  13 +-
src/test/java/net/sourceforge/plantuml/servlet/TestMap.java         |  80 +++++---
src/test/java/net/sourceforge/plantuml/servlet/TestOldProxy.java    |  84 ++++-----
src/test/java/net/sourceforge/plantuml/servlet/TestProxy.java       | 120 ++++++++----
src/test/java/net/sourceforge/plantuml/servlet/TestSVG.java         | 131 +++++++------
src/test/java/net/sourceforge/plantuml/servlet/TestUtils.java       |   6 +-
src/test/java/net/sourceforge/plantuml/servlet/WebappTestCase.java  |  81 +++++++-
18 files changed, 660 insertions(+), 496 deletions(-)
```

The first 4 files contain real changes.
The test files themselfe should not contain important changes (framework switch).


### Changes

- Test Server
  * It is now possible run the junit tests against an external plantuml-server.
    This could be used to also test the finished jetty und tomcat docker images.
    Note: `ServerUtils.java` moved to `server/EmbeddedJettyServer.java`
  * **Open TODO**: The embedded jetty server (before ServerUtils.java) uses jetty 8. Jetty 8 is deprecated and at least I have sometimes problems to get this version running on my system.
    Possible solutions
      - update to jetty 11
      - test against external server, e.g. `docker-compose up` (also see: WebappTestCase Constructor)
- JUnit Tests
  * Simplify tests by using `java.net.URL` instead of an (old) framework.
  * Enable JavaScript by default for a more realistic testing environment.
  * Some little fixes
- pom.xml
  * remove [HTTPClient artifact](https://mvnrepository.com/artifact/HTTPClient/HTTPClient)
    HTTPClient was last updated in Nov, 2005
    This was only used in `OldProxyServlet.handleImageProxy()`
  * switch from [HttpUnit](https://mvnrepository.com/artifact/httpunit/httpunit) to [HtmlUnit](https://mvnrepository.com/artifact/net.sourceforge.htmlunit/htmlunit)
    HttpUnit is an old framework which is nearly never updates
  * remove [rhino:js](https://mvnrepository.com/artifact/rhino/js)
    Only necessary in combination with `HttpUnit`.
  * Make tests at least manually runable.
    Currently, test will be skipped by default. Add `skipTests` as property so that you can at least manually change it via:
    - `mvn test -DskipTests=false`
    - `mvn test -DskipTests=false -DargLine="-Dsystem.test.server=http://localhost:8080/plantuml"`
